src/main/java/uk/nhs/digital/nhsconnect/nhais/mapper/TransactionTypeMapper.java

```diff
@@ -0,0 +1,20 @@
+package uk.nhs.digital.nhsconnect.nhais.mapper;
+
+import uk.nhs.digital.nhsconnect.nhais.exceptions.ParameterValidationException;
+import uk.nhs.digital.nhsconnect.nhais.model.edifact.ReferenceTransactionType;
+
+import org.springframework.stereotype.Component;
+
+@Component
+public class TransactionTypeMapper {
+
+    public ReferenceTransactionType.Outbound mapTransactionType(String transactionTypeParam) {
+        switch(transactionTypeParam) {
+            case "$nhais.acceptance": return ReferenceTransactionType.Outbound.ACCEPTANCE;
+            case "$nhais.removal": return ReferenceTransactionType.Outbound.REMOVAL;
+            case "$nhais.deduction": return ReferenceTransactionType.Outbound.DEDUCTION;
+            default: throw new ParameterValidationException("Provided transaction type is not allowed");
```

Comment:
It would be helpful to include the operation:

`"Unknown Patient operation {}", transactionTypeParam`

transactionTypeParam actually refers to a FHIR Operation https://fhir-drills.github.io/operations.html which is why we use this $ prefix